Fix #1646:Changed all padding_# and margin_# to space_#dp in dimens value.

ranjsa opened the pull request:
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes [#1646 ](https://github.com/oppia/oppia-android/issues/1646)

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

ranjsa committed:
fix-dimens-value-to-follow-one-naming-convention

ranjsa committed:
fixed max line length error

anandwana001 commented on app/src/main/res/layout-land/walkthrough_final_fragment.xml:
```diff
@@ -43,12 +43,12 @@
         android:onClick="@{(v) -> presenter.goBack()}"
         app:cardCornerRadius="4dp"
-        app:contentPadding="@dimen/padding_24dp">
+        app:contentPadding="@dimen/space_24dpdp">
```
it should be `space_24dp` not `dpdp`. 

here and all places.

ranjsa replied:
done

anandwana001 commented on app/src/main/res/layout-land/walkthrough_final_fragment.xml:
```diff
@@ -75,13 +75,13 @@
         android:focusable="true"
         app:cardCornerRadius="4dp"
-        app:contentPadding="@dimen/padding_24dp">
+        app:contentPadding="@dimen/space_24dpdp">
```
ditto

ranjsa replied:
done

anandwana001 commented on app/src/main/res/layout/walkthrough_final_fragment.xml:
```diff
@@ -48,12 +48,12 @@
           android:onClick="@{(v) -> presenter.goBack()}"
           app:cardCornerRadius="4dp"
-          app:contentPadding="@dimen/padding_24dp">
+          app:contentPadding="@dimen/space_24dpdp">
```
ditto

ranjsa replied:
done

anandwana001 commented on app/src/main/res/layout/walkthrough_final_fragment.xml:
```diff
@@ -80,13 +80,13 @@
           android:focusable="true"
           app:cardCornerRadius="4dp"
-          app:contentPadding="@dimen/padding_24dp">
+          app:contentPadding="@dimen/space_24dpdp">
```
ditto

anandwana001 commented on app/src/main/res/values/dimens.xml:
```diff
@@ -14,58 +13,42 @@
   <dimen name="elevation_4dp">4dp</dimen>
   <dimen name="elevation_0dp">0dp</dimen>
   <dimen name="padding_296dp">296dp</dimen>
-  <dimen name="padding_72dp">72dp</dimen>
```
@rt4914 Do we have any specific difference in `padding_72dp ` vs `padding_72 `?

anandwana001 reviewed (changes requested):
Nice work @ranjsa 
Let's wait for Rajat comment on dpdp thing and then you can update it accordingly. 

Added a few points and also added 1 point to clarify between `padding_72` vs `padding_72dp`.

ranjsa committed:
fixed space_#dpdp to space_#dp

ranjsa commented:
> Nice work @ranjsa
> Let's wait for Rajat comment on dpdp thing and then you can update it accordingly.
> 
> Added a few points and also added 1 point to clarify between `padding_72` vs `padding_72dp`.

Thank You @anandwana001 for reviewing my PR. I have fixed all space_#dpdp to space_#dp.

anandwana001 commented on app/src/main/res/values/dimens.xml:
```diff
@@ -14,58 +13,40 @@
   <dimen name="elevation_4dp">4dp</dimen>
   <dimen name="elevation_0dp">0dp</dimen>
   <dimen name="padding_296dp">296dp</dimen>
```
let's update this padding too.

anandwana001 replied:
@ranjsa I think we are not using this padding anywhere, could you please verify this and remove if not needed?

ranjsa replied:
yes @anandwana001 we were not using this anywhere so I removed it

ranjsa committed:
removed unused padding_296dp

anandwana001 reviewed (approved):
LGTM, Thanks @ranjsa 
@rt4914 can take a quick pass on this and then we are good to merge this. By the time, feel free to check other issues to work on.

ranjsa commented:
> LGTM, Thanks @ranjsa
> @rt4914 can take a quick pass on this and then we are good to merge this. By the time, feel free to check other issues to work on.

Okay sure, Thanks @anandwana001

BenHenning commented:
Ideally, we should be semantically naming all dimensions. @rt4914 would you prefer that we check this PR in as-is and fix that in a later change?

rt4914 reviewed (approved):
Thanks  a lot @ranjsa . This is really great.

Also, I understand that its not easy to make a change and resolve merge conflicts again and again and therefore I am approving this so that if you want to merge this PR at this stage you can do that.

If you choose to make the suggested change, then just assign it back to me and I review back and merge.

If you choose to merge it right now, then I suggest you file an issue [here](https://github.com/oppia/oppia-android/issues/new) . You can also add this under SLoP tags too.

Sounds good?

rt4914 commented:
> Ideally, we should be semantically naming all dimensions. @rt4914 would you prefer that we check this PR in as-is and fix that in a later change?

@BenHenning I am not 100% sure what you mean here?
(a.) Is it about reordering or the `space_` values in `dimens.xml` file
or (b.) The naming should be changed to provide more context to paddings/margins?
or something else.

ranjsa commented:
Hello, @rt4914 thanks you for reviewing this PR.
Also, did you mean  creating a new issue for this https://github.com/oppia/oppia-android/pull/1936#issuecomment-704559424 
https://github.com/oppia/oppia-android/pull/1936#issuecomment-704996261 if so I would like to work on this issue on a different PR if I am right then please confirm once so I can proceed.

rt4914 commented:
> Hello, @rt4914 thanks you for reviewing this PR.
> Also, did you mean creating a new issue for this [#1936 (comment)](https://github.com/oppia/oppia-android/pull/1936#issuecomment-704559424)
> [#1936 (comment)](https://github.com/oppia/oppia-android/pull/1936#issuecomment-704996261) if so I would like to work on this issue on a different PR if I am right then please confirm once so I can proceed.

Yeah, you should file a new issue.
I will merge this PR now and you can file a new issue and start working on it.
